### PR TITLE
feat: Implement graceful shutdown

### LINKED ISSUE #20: Add graceful shutdown in DropKit
Refer to the following links for more information.
* https://blog.wu-boy.com/2020/02/what-is-graceful-shutdown-in-golang/
* https://callistaenterprise.se/blogg/teknik/2019/10/05/go-worker-cancellation/

### PR DESCRIPTION
**Pull request by howjmay**
Close #20

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,6 +32,19 @@ func init() {
 }
 
 func main() {
-	router := routes.SetupRouter()
-	router.Run(":" + viper.GetString(`DROPKIT.PORT`))
+	// router := routes.SetupRouter()
+	_, srv := routes.SetupRouter()
+	// router.Run(":" + viper.GetString(`DROPKIT.PORT`))
```
Comment (jserv) [inline main.go:37]: Don't leave the dead code.

Comment (jserv) [reply main.go:37]: ?

File: main.go [hunk 2]
```diff
@@ -25,6 +32,19 @@ func init() {
...
+
+	shutdown := make(chan os.Signal)
+	signal.Notify(shutdown, syscall.SIGINT, syscall.SIGTERM)
+	<-shutdown
+	log.Info("Shutdown Server")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (jserv) [inline main.go:44]: Why did you set `5` times here?

Comment (howjmay) [reply main.go:44]: the 5 here means this context (a series of dependent stacks) would be closed in 5 sec

Comment (jserv) [reply main.go:44]: You have to document it.

**Pull request closed without merging**